File: src/main/java/com/braydenoneal/blang/parser/expression/value/Null.java
package com.braydenoneal.blang.parser.expression.value;

import com.mojang.serialization.Codec;

public record Null() {
    public static final Codec<Null> CODEC = Codec.unit(new Null());

    public static NullValue value() {
        return new NullValue(new Null());
    }
}

File: src/main/kotlin/com/braydenoneal/blang/parser/expression/value/Null.kt
package com.braydenoneal.blang.parser.expression.value

import com.braydenoneal.blang.parser.Program
import com.braydenoneal.blang.parser.expression.Expression
import com.mojang.serialization.Codec

class Null {
    companion object {
        val CODEC: Codec<Null> = Codec.unit<Null>(Null())
        val VALUE = NullValue(Null())

        fun parse(program: Program): Expression {
            program.next()
            return VALUE
        }
    }
}
